src/data/finsupp/basic.lean

```diff
@@ -1465,6 +1505,20 @@ lemma to_multiset_apply (f : α →₀ ℕ) : f.to_multiset = f.sum (λ a n, n 
 @[simp] lemma to_multiset_single (a : α) (n : ℕ) : to_multiset (single a n) = n •ℕ {a} :=
 by rw [to_multiset_apply, sum_single_index]; apply zero_nsmul
 
+lemma to_multiset_sum {ι : Type*} {f : ι → α →₀ ℕ} (s : finset ι) :
+  finsupp.to_multiset (∑ i in s, f i) = ∑ i in s, finsupp.to_multiset (f i) :=
+begin
+  apply finset.induction_on s,
+  { simp },
+  { intros i s hi,
+    simp [hi] }
+end
+
+lemma to_multiset_sum_single {ι : Type*} (s : finset ι) (n : ℕ) :
+  finsupp.to_multiset (∑ i in s, single i n) = n • s.val :=
+by simp_rw [to_multiset_sum, finsupp.to_multiset_single, multiset.singleton_eq_singleton,
+            sum_nsmul, sum_multiset_singleton, nsmul_eq_smul]
```

Comment:
I'd be inclined to leave this as `nsmul`, converting to `nsmul` here just increases the chance of diamonds.